Building tensorflow (makefile method) on arhmf Olimex A20

Building tensorflow using the makefile method, i get the below error, 

`PROTOC = "protoc"
CC_PREFIX = ""
gcc --std=c++11 -DIS_SLIM_BUILD -O0 -I/usr/local/include -I. -I/home/olimex/tf/tensorflow/tensorflow/contrib/makefile/downloads/ -I/home/olimex/tf/tensorflow/tensorflow/contrib/makefile/downloads/eigen-eigen-334b1d428283 -I/home/olimex/tf/tensorflow/tensorflow/contrib/makefile/gen/proto/ -I/home/olimex/tf/tensorflow/tensorflow/contrib/makefile/gen/proto_text/ \
        -o /home/olimex/tf/tensorflow/tensorflow/contrib/makefile/gen/bin/benchmark /home/olimex/tf/tensorflow/tensorflow/contrib/makefile/gen/obj/tensorflow/core/util/reporter.o /home/olimex/tf/tensorflow/tensorflow/contrib/makefile/gen/obj/tensorflow/tools/benchmark/benchmark_model.o /home/olimex/tf/tensorflow/tensorflow/contrib/makefile/gen/obj/tensorflow/tools/benchmark/benchmark_model_main.o \
         -Wl,--allow-multiple-definition -Wl,--whole-archive /home/olimex/tf/tensorflow/tensorflow/contrib/makefile/gen/lib/libtensorflow-core.a -L/usr/local/lib -lstdc++ -lprotobuf -lz -lm -ldl -lpthread
/usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o): In function `__gnu_h2f_ieee':
(.text+0x12a): relocation truncated to fit: R_ARM_THM_JUMP11 against symbol `__gnu_h2f_internal' defined in .text section in /usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o)
/usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o): In function `__gnu_h2f_alternative':
(.text+0x132): relocation truncated to fit: R_ARM_THM_JUMP11 against symbol `__gnu_h2f_internal' defined in .text section in /usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o)
/usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o): In function `__gnu_h2f_ieee':
(.text+0x12a): relocation truncated to fit: R_ARM_THM_JUMP11 against symbol `__gnu_h2f_internal' defined in .text section in /usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o)
/usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o): In function `__gnu_h2f_alternative':
(.text+0x132): relocation truncated to fit: R_ARM_THM_JUMP11 against symbol `__gnu_h2f_internal' defined in .text section in /usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o)
/usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o): In function `__gnu_h2f_ieee':
(.text+0x12a): relocation truncated to fit: R_ARM_THM_JUMP11 against symbol `__gnu_h2f_internal' defined in .text section in /usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o)
/usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o): In function `__gnu_h2f_alternative':
(.text+0x132): relocation truncated to fit: R_ARM_THM_JUMP11 against symbol `__gnu_h2f_internal' defined in .text section in /usr/lib/gcc/arm-linux-gnueabihf/4.8/libgcc.a(fp16.o)
collect2: error: ld returned 1 exit status`

From what i understand this appears to be an issue with the linker trying to branch on a long address, 
https://lists.gnu.org/archive/html/bug-binutils/2007-06/msg00065.html
But this issue is back from 2007, any workaround i can do on the source to avoid this?

System:
gcc version 4.8.4 (Debian 4.8.4-1)
GNU ld (GNU Binutils for Debian) 2.25

--- a/tensorflow/contrib/makefile/Makefile
+++ b/tensorflow/contrib/makefile/Makefile
@@ -188,7 +188,8 @@ ifeq ($(HAS_GEN_HOST_PROTOC),true)
 	LIBFLAGS += -L$(MAKEFILE_DIR)/gen/protobuf-host/lib
 	export LD_LIBRARY_PATH=$(MAKEFILE_DIR)/gen/protobuf-host/lib
 endif
-	LIBFLAGS += -Wl,--allow-multiple-definition -Wl,--whole-archive
+	LIBFLAGS += -Wl,--whole-archive
+	LDFLAGS := -Wl,--no-whole-archive
 endif
 # If we're on Linux, also link in the dl library.
 ifeq ($(TARGET),LINUX)

--- a/tensorflow/contrib/makefile/Makefile
+++ b/tensorflow/contrib/makefile/Makefile
@@ -188,7 +188,7 @@ ifeq ($(HAS_GEN_HOST_PROTOC),true)
 	LIBFLAGS += -L$(MAKEFILE_DIR)/gen/protobuf-host/lib
 	export LD_LIBRARY_PATH=$(MAKEFILE_DIR)/gen/protobuf-host/lib
 endif
-	LIBFLAGS += -Wl,--whole-archive
+	LIBFLAGS += -Wl,--allow-multiple-definition -Wl,--whole-archive
 	LDFLAGS := -Wl,--no-whole-archive
 endif
 # If we're on Linux, also link in the dl library.

